cognite/powerops/client/data_classes/shop_result_files.py

```diff
@@ -74,15 +73,8 @@ def file_content(self) -> str:
     def _retrieve_time_series_dict(self, key: str) -> dict[datetime, float]:
         # key is a dot separated string of nested keys
         try:
-            # FIXME? This is a workaround for the nested keys without dot get and
-            # the fact that the yaml parser parses numeric keys as numbers
-            keys = [int(k) if k.isdigit() else k for k in key.split(".")]
-            data = reduce(operator.getitem, keys, self.data)
-            if not (
-                isinstance(data, dict)
-                and all(isinstance(k, datetime) for k in data.keys())
-                and all(isinstance(v, (float, int)) for v in data.values())
-            ):
+            data = get_dict_dot_keys(self.data, key)
+            if not (is_time_series_dict(data)):
```

Comment:
Don't think the extra parathesis 
```suggestion
            if not is_time_series_dict(data):
```